lib/trento/application/integration/discovery/protocol/enrich_request_host_deregistration.ex

```diff
@@ -0,0 +1,47 @@
+defimpl Trento.Support.Middleware.Enrichable,
+  for: Trento.Domain.Commands.RequestHostDeregistration do
+  alias Trento.Domain.Commands.RequestHostDeregistration
+
+  alias Trento.Repo
+
+  alias Trento.Heartbeat
+  alias Trento.HostReadModel
+
+  import Ecto.Query
+
+  @heartbeat_interval Application.compile_env!(:trento, Trento.Heartbeats)[:interval]
+  @deregistration_debounce Application.compile_env!(
+                             :trento,
+                             :deregistration_debounce
+                           )
+  @total_deregistration_debounce @heartbeat_interval + @deregistration_debounce
+
+  @spec enrich(RequestHostDeregistration.t(), map) ::
+          {:ok, RequestHostDeregistration.t()}
+          | {:error, :host_alive}
+          | {:error, :host_not_registered}
+  def enrich(%RequestHostDeregistration{host_id: host_id} = command, _) do
+    host_exists = Repo.exists?(HostReadModel, id: host_id)
```

Comment:
Please short circuit here, we don't need to perform the query below, if the host does not exists

You can use different functions, or maybe a with - else block, but it's better if we avoid to perform useless query, not perfomance wise obvs, the query is really simple, but for general correctness